### PR TITLE
implement Scanner for Many and Branch

### PR DESCRIPTION
**Pull request by ChloePlanet**
Changes:
- add Merge function to merge multiple same-src Scanners
- implement Scanner function for `Many` and `Branch`
- bold context filename text and show oneline context by default

### CODE REVIEW

File: ast/node_string.go [hunk 1]
```diff
@@ -115,7 +115,18 @@ func (c One) Scanner() parser.Scanner {
 }
 
 func (c Many) Scanner() parser.Scanner {
-	panic("Scanner() not valid for Many")
+	childrenScanners := make([]parser.Scanner, len(c))
+	for i, n := range c {
+		childrenScanners[i] = n.Scanner()
+	}
```
Comment (marcelocantos) [inline ast/node_string.go:121]: nit: `make([]parser.Scanner, 0, len(c)) ... append` is more idiomatic.

File: parser/scanner.go [hunk 2]
```diff
@@ -118,6 +119,33 @@ func (s Scanner) Skip(i int) *Scanner {
 	return &Scanner{s.src, s.sliceStart + i, s.sliceLength - i}
 }
 
+func (s *Scanner) Merge(items []Scanner) error {
+	if len(items) < 2 {
+		return errors.New("merge needs at least two scanners")
```
Comment (marcelocantos) [inline parser/scanner.go]: `len(items) == 1` is a degenerate case, but still valid.

File: parser/scanner.go [hunk 3]
```diff
@@ -118,6 +119,33 @@ func (s Scanner) Skip(i int) *Scanner {
 	return &Scanner{s.src, s.sliceStart + i, s.sliceLength - i}
 }
 
+func (s *Scanner) Merge(items []Scanner) error {
```
Comment (marcelocantos) [inline parser/scanner.go]: Prefer a more functional approach:
```suggestion
func MergeScanners(items ...Scanner) (Scanner, error) {
```

**Review by marcelocantos — CHANGES_REQUESTED** [review]

**Review by marcelocantos — DISMISSED** [review]

**Review by marcelocantos — DISMISSED** [review]
One or two negative tests (no leaves) would be good.

**Review by anzdaddy — DISMISSED** [review]

**Review by marcelocantos — APPROVED** [review]

**Pull request merged**